cuda_bindings/cuda/bindings/_path_finder/load_dl_common.py

```diff
@@ -38,3 +40,21 @@ def load_dependencies(libname: str, load_func: Callable[[str], LoadedDL]) -> Non
     """
     for dep in DIRECT_DEPENDENCIES.get(libname, ()):
         load_func(dep)
+
+
+def load_in_subprocess(python_code, timeout=30):
+    # This is to avoid loading libraries into the parent process.
+    return subprocess.run(  # nosec B603
+        [sys.executable, "-c", python_code],
+        capture_output=True,
+        encoding="utf-8",
+        timeout=timeout,  # Ensure this does not hang for an excessive amount of time.
+    )
```

Comment:
I am juggling between chats so just leaving a drive-by comment for now. We need to pass `env=os.environ` too to ensure the new process is seeing the same environment (`PATH`, `LD_LIBRARY_PATH`, `CONDA_PREFIX`, etc).

Reply:
Done: commit 1f728c0cb4a5a99788e230edb739cc29a12ded7a

Thanks for catching this!